### PR TITLE
Net worth calculator

### PR DESCRIPTION
**Pull request by DLi2004**
Creates a function to calculate the net worth of a user (account balance + portfolio value + value of current orders)

### CODE REVIEW

**Comment by evanFroelich** [comment]
before merging, take a pull from main and settle the changes. We removed several supabase middleman functions that I see in your PR

**Comment by DLi2004** [comment]
@evanFroelich The only middleman function I believe we use in networth_calculator is fetch_stock_price, which I don't think was changed. Should we incorporate the changes to the other functions in the file or are there any changes we should make to the networth_calculator function?

File: backend/routes/net_worth_calculator.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from supabase import Client, create_client
+
+from database import supabase_middleman
+
+load_dotenv("env/.env")
+
+
+url = os.getenv("PUBLIC_SUPABASE_URL")
+key = os.getenv("SUPABASE_KEY")
+supabase: Client = create_client(url, key)
```
Comment (evanFroelich) [inline backend/routes/net_worth_calculator.py:16]: ideally only supabase_middleman handles db transactions

File: backend/routes/net_worth_calculator.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    Gets the user's networth from profile, portfolio, and active_buy_sell
+
+    Args:
+        user_id (int): The id of the user whose net worth is requested
+
+    Returns: Net worth of user
+    """
+
+    # Get the user's balance
+    profile_balance = (supabase.table("profiles")
```
Comment (evanFroelich) [inline backend/routes/net_worth_calculator.py:30]: turn into a generic middleman function

File: backend/routes/net_worth_calculator.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        .select("balance")
+        .match({"userId": user_id})
+        .execute()
+        .data
+        .pop()["balance"])
+
+    # Get the user's portfolio
+    user_portfolio = (supabase.table("portfolio")
```
Comment (evanFroelich) [inline backend/routes/net_worth_calculator.py:38]: turn into a generic middleman function

File: backend/routes/net_worth_calculator.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        .execute()
+        .data)
+    portfolio_balance = 0
+
+    # Calculate the value of the user's portfolio
+    for stock in user_portfolio:
+        portfolio_balance += stock["quantity"] * supabase_middleman.fetch_stock_price(stock["stockId"])
+
+    # Get the user's active buy/sell orders
+    active_buy_sell_entries = (supabase.table("active_buy_sell")
```
Comment (evanFroelich) [inline backend/routes/net_worth_calculator.py:50]: turn into a generic middleman function

**Review by evanFroelich — APPROVED** [review]
looks good

**Pull request merged**